conn.go

```diff
@@ -4,23 +4,36 @@ import (
 	"fmt"
 	"io"
 	"net"
+	"time"
 )
 
 type transmitFunc func(buf []byte) (bool, error)
 
 type Conn struct {
-	addr     net.Addr
-	conn     net.PacketConn
-	protocol *Protocol
+	addr net.Addr
+	conn net.PacketConn
+
+	p *Protocol
+
+	updatePeriod  time.Duration // how often time-dependant parts of the p get checked
+	resendTimeout time.Duration // how long we wait until unacked packets should be resent
+
+	exit chan struct{} // signal channel to close the conn
 }
 
 func NewConn(addr net.Addr, conn net.PacketConn, opts ...ProtocolOption) *Conn {
-	p := NewProtocol(opts...)
-	return &Conn{addr: addr, conn: conn, protocol: p}
+	return &Conn{
+		addr:          addr,
+		conn:          conn,
+		p:             NewProtocol(opts...),
+		updatePeriod:  DefaultUpdatePeriod,
+		resendTimeout: DefaultResendTimeout,
+		exit:          make(chan struct{}),
+	}
```

Comment:
warning: exported method Conn.WriteUnreliablePacket should have comment or be unexported (golint)